lib/web/integrations_awsoidc.go

```diff
@@ -280,19 +280,29 @@ func (h *Handler) awsOIDCListDeployedDatabaseService(w http.ResponseWriter, r *h
 		return nil, trace.Wrap(err)
 	}
 
-	services, err := listDeployedDatabaseServices(ctx, h.logger, integrationName, regions, clt.IntegrationAWSOIDCClient())
+	if len(regions) == 0 {
+		// return an empty list if there are no relevant regions in which to fetch database services
+		return ui.AWSOIDCListDeployedDatabaseServiceResponse{}, nil
+	}
+
+	s, err := listDeployedDatabaseServices(ctx, h.logger, integrationName, regions, clt.IntegrationAWSOIDCClient())
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
 	return ui.AWSOIDCListDeployedDatabaseServiceResponse{
-		Services: services,
+		Services: s,
 	}, nil
 }
 
 func extractAWSRegionsFromQuery(r *http.Request) ([]string, error) {
 	var ret []string
 	for _, region := range r.URL.Query()["regions"] {
+		if region == "" {
+			// no regions passed in params, empty key
+			return nil, nil
```

Comment:
Nit: Returning `nil, nil` is generally frowned-upon, I would return `ret` here to more explicitly indicate we're returning empty regions list (even if it's technically the same in this case).
```suggestion
			return ret, nil
```